Sources/StreamChat/Models/Attachments/ChatMessageFileAttachment.swift

```diff
@@ -22,6 +22,13 @@ public struct FileAttachmentPayload: AttachmentPayload {
     public internal(set) var assetURL: URL
     /// The file itself.
     public let file: AttachmentFile
+    
+    let _extraData: [String: RawJSON]
+    
+    /// An extra data.
+    public var extraData: [String: Any] {
+        _extraData.compactMapValues(\.value)
+    }
```

Comment:
We could make it smarter and expose generic API that would decode from `[String: RawJSON]`. 

Smth like:
```swift
public func extraData<T: Decodable>() -> T? {
 // try to decode `T` from [String: Any] 
}
```

Let me know if it worth trying